src/nvim/os/env.c

```diff
@@ -427,6 +427,33 @@ char *vim_getenv(const char *name)
     return xstrdup(kos_env_path);
   }
 
+  // XDG Base Directory expansion.
+  if (strncmp(name, "XDG_", 4) == 0) {
+    const char *home = vim_getenv("HOME");
+    if (!home) {
+        return NULL;
+    }
+
+    const char *home_suffix = "";
+    if (strcmp(name, "XDG_DATA_HOME") == 0) {
+      home_suffix = "/.local/share/";
+    } else if (strcmp(name, "XDG_CONFIG_HOME") == 0) {
+      home_suffix = "/.config/";
+    } else if (strcmp(name, "XDG_CACHE_HOME") == 0) {
+      home_suffix = "/.cache/";
+    } else if (strcmp(name, "XDG_RUNTIME_DIR") == 0) {
+      return xstrdup("/tmp/");  // Arbitrary value, any idea?
+    }
```

Comment:
You are right. It's just that it's a bigger change than simply using *_HOME, since you can't expand XDG_*_DIR in macros, it needs code changes.